src/main/java/bio/terra/janitor/app/controller/UnauthenticatedApiController.java

```diff
@@ -17,27 +18,48 @@
 @Controller
 public class UnauthenticatedApiController implements UnauthenticatedApi {
   private final NamedParameterJdbcTemplate jdbcTemplate;
+  private final StairwayComponent stairwayComponent;
 
   @Autowired
-  UnauthenticatedApiController(JanitorJdbcConfiguration jdbcConfiguration) {
+  UnauthenticatedApiController(
+      JanitorJdbcConfiguration jdbcConfiguration, StairwayComponent stairwayComponent) {
     this.jdbcTemplate = new NamedParameterJdbcTemplate(jdbcConfiguration.getDataSource());
+    this.stairwayComponent = stairwayComponent;
   }
 
   @Override
   public ResponseEntity<SystemStatus> serviceStatus() {
-    if (jdbcTemplate.getJdbcTemplate().execute((Connection connection) -> connection.isValid(0))) {
-      return new ResponseEntity<>(
-          new SystemStatus()
-              .ok(true)
-              .putSystemsItem("postgres", new SystemStatusSystems().ok(true)),
-          HttpStatus.OK);
+    SystemStatus systemStatus = new SystemStatus();
+
+    final boolean postgresOk =
+        jdbcTemplate.getJdbcTemplate().execute((Connection connection) -> connection.isValid(0));
+    systemStatus.putSystemsItem("postgres", new SystemStatusSystems().ok(postgresOk));
+
+    StairwayComponent.Status stairwayStatus = stairwayComponent.getStatus();
```

Comment:
I thought "subSystem" would be the "systems" janitor depends. And the one above already checks postgress. This one you added also checks postgres by using another db. And the thing you added also sounds like more than postgres healthy, it also includes "is db ready to use". Kind of this has overlaps with the previous one and we may need a clear definition about what level of scope the "subSystem" we want here.  An example in WM: https://github.com/DataBiosphere/terra-workspace-manager/blob/b19234a73a977898d0b77ac359f38631d71e5d3a/src/main/java/bio/terra/workspace/service/status/WorkspaceManagerStatusService.java

Reply:
Hmmm, yeah. I think at the end of the day the status should only return ok if the janitor service can operate, i.e. it's healthy. For stairway, that includes the database connection but also (soon, but not yet) a pubsub topic/subscription existing. I'd kind of like to leave those details of what Stairway needs to Stairway instead of introspecting it. I'm not sure how strictly to define a "subsystem." Being a little fuzzy about it seems ok to me.